csharp/src/Ice/Ice1NetworkSocket.cs

```diff
@@ -289,7 +290,9 @@ async ValueTask QueueAsync(IList<ArraySegment<byte>> buffer, CancellationToken c
                     // Wait for the previous send to complete
                     await _sendTask.ConfigureAwait(false);
                 }
-                catch (DatagramLimitException)
+                catch (TransportException ex) when (Endpoint.IsDatagram &&
+                                                    ex.InnerException is SocketException socketException &&
+                                                    socketException.SocketErrorCode == SocketError.MessageSize)
                 {
                     // If the send failed because the datagram was too large, ignore and continue sending.
```

Comment:
Should be "if the _previous_ send failed". 
When a Send/SendAsync fails, we presumably notify the sender with an exception. However, the next send(er) should not be affected.

Reply:
Yes, the sender waits on the send task and will get the exception.